mart/attack/adversary.py

```diff
@@ -275,14 +264,24 @@ def forward(
         model: Optional[torch.nn.Module] = None,
         **kwargs
     ):
+        # Backward compatible: earlier configs have two unnamed arguments [input, target].
+        kwargs["input"] = input
+        kwargs["target"] = target
+        kwargs["model"] = model
+
+        # The benign input to an adversary is not necessarily the model input.
+        # It could be activations of other modules.
+        self.benign_input = kwargs[self.benign_input_key]
```

Comment:
Why is this necessary? I don't quite understand the comment. What is the "model input" in the context of an "Adversary"? This definitely needs more explanation or rethinking.

Reply:
Like what is the distinction between benign input and input?